src/main/kotlin/interpreter/pyobject/PyObject.kt

```diff
@@ -109,31 +117,26 @@ abstract class PyObject() {
      * @param name: The name of the attribute to get.
      */
     open fun pyGetAttribute(name: String): PyObject {
-        // this will delegate to `__getattribute__`,
-
-        // try and run __getattribute__
-        val getAttribute = try {
-            ObjectGetattribute.runCallable(
-                listOf(PyString("__getattribute__"), this)
-            )
-        } catch (e: KyError) {
-            ObjectGetattribute.pyDescriptorGet(this, type)
+        // try and find a magic method
+        val magicMethod = magicSlots.nameToMagicMethodBound(this, name)
+        if (magicMethod != null) {
```

Comment:
`let` is exactly meant for this without making it too functional. You're also not re-using the var.
```kotlin
magicSlots.nameToMagicMethodBound(this, name)?.let { 
            return it as PyObject
}
```